### PR TITLE
Fix reorg depth count 

### LINKED ISSUE #5136: Reorg depth counter is incorrect
**Describe the bug**

Recent reorgs are not properly being shown in our metrics. @wemeetagain investigated why our metrics cannot properly show and calculate reorgs and their depths. Reorg depth is currently calculated as `newHead.slot - commonAncestor.slot` 

**Expected behavior**

The reorg depth counter should be able to determine reorgs and at what depth. We need to tweak the calculation with something else. @wemeetagain suggests maybe `Math.max(newHead.slot, oldHead.slot) - commonAncestor.slot` 

Open for discussion/solutions.

### PR DESCRIPTION
**Pull request by maschad**
Closes https://github.com/ChainSafe/lodestar/issues/5136

### CODE REVIEW

File: packages/fork-choice/src/forkChoice/forkChoice.ts [hunk 1]
```diff
@@ -786,7 +786,7 @@ export class ForkChoice implements IForkChoice {
       return {code: AncestorStatus.Descendant};
     }
 
-    return {code: AncestorStatus.CommonAncestor, depth: newNode.slot - commonAncestor.slot};
+    return {code: AncestorStatus.CommonAncestor, depth: Math.max(newNode.slot, prevNode.slot) - commonAncestor.slot};
```
Comment (wemeetagain) [inline packages/fork-choice/src/forkChoice/forkChoice.ts:789]: Can you update the tsdoc for the function? may also need to be updated in the interface definition

File: packages/fork-choice/src/forkChoice/forkChoice.ts [hunk 2]
```diff
@@ -767,7 +767,7 @@ export class ForkChoice implements IForkChoice {
     return blocksAtSlot;
   }
 
-  /** Returns the distance of common ancestor of nodes to newNode. Returns null if newNode is descendant of prevNode */
+  /** Returns the distance of common ancestor of nodes to the max of the newNode and the prevNode. Returns null if newNode is descendant of prevNode */
```
Comment (wemeetagain) [inline packages/fork-choice/src/forkChoice/forkChoice.ts]: Seems this null comment is stale. Can you just remove it?

Comment (maschad) [reply packages/fork-choice/src/forkChoice/forkChoice.ts]: Line 781 suggests the noCommonAncestor error code is equivalent to a null response which is why I left it

Comment (wemeetagain) [reply packages/fork-choice/src/forkChoice/forkChoice.ts]: I think that comment was from before we returned a `AncestorResult`.

Comment (wemeetagain) [inline packages/fork-choice/src/forkChoice/forkChoice.ts]:
```suggestion
  /** Returns the distance of common ancestor of nodes to the max of the newNode and the prevNode. */
```

File: packages/fork-choice/src/forkChoice/interface.ts [hunk 3]
```diff
@@ -177,7 +177,7 @@ export interface IForkChoice {
   forwardIterateDescendants(blockRoot: RootHex): IterableIterator<ProtoBlock>;
   getBlockSummariesByParentRoot(parentRoot: RootHex): ProtoBlock[];
   getBlockSummariesAtSlot(slot: Slot): ProtoBlock[];
-  /** Returns the distance of common ancestor of nodes to newNode. Returns null if newNode is descendant of prevNode */
+  /** Returns the distance of common ancestor of nodes to the max of the newNode and the prevNode. Returns null if newNode is descendant of prevNode */
```
Comment (wemeetagain) [inline packages/fork-choice/src/forkChoice/interface.ts]:
```suggestion
  /** Returns the distance of common ancestor of nodes to the max of the newNode and the prevNode. */
```

**Review by wemeetagain — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.6.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.6.0) :tada:

**Pull request merged**